sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/BodyMemory.cs

```diff
@@ -33,20 +34,30 @@ private ReadOnlyMemory<byte> WrittenMemory
         }
 
         // for the send path the combined memory for the body is not precomputed
-        public BodyMemory(IEnumerable<ReadOnlyMemory<byte>> dataSegments)
+        private BodyMemory(IEnumerable<ReadOnlyMemory<byte>> dataSegments)
         {
             _lazySegments = dataSegments;
         }
 
         // for the receive path the combined memory is precomputed because we need to copy the underlying AMQP data anyway
-        public BodyMemory(IEnumerable<Data> dataSegments)
+        private BodyMemory(IEnumerable<Data> dataSegments)
         {
             foreach (var segment in dataSegments)
             {
                 Append(segment);
             }
         }
 
+        public static BodyMemory From(IEnumerable<ReadOnlyMemory<byte>> segments)
+        {
+            return segments as BodyMemory ?? new BodyMemory(segments);
+        }
+
+        public static BodyMemory From(IEnumerable<Data> segments)
```

Comment:
nit: it might be helpful to give these factory methods different names, like FromAmqpData, and FromReadOnlyMemory.

Reply:
Done